src/opencmiss/zincwidgets/materialeditorwidget.py

```diff
@@ -0,0 +1,382 @@
+            # itemDelegate.commitData(self._onMaterialItemChanged)
+            self._ui.materials_listView.setItemDelegate(itemDelegate)
+            self._ui.materials_listView.show()
+            self._displayMaterial()
+
+    def _onMaterialItemChanged(self, item):
+        """
+        For QStandardItemModel ItemChanged Signal, catch the item text edit event.
+        Update the material name.
+        """ 
+        newName = item.text()
+        if self._renameMaterial(item.data(), newName):
+            return
+        item.setText(item.data().getName())
+        QtWidgets.QMessageBox.information(self, "Info", "Can't change this material's name to %s."%newName)
+
+    def _displayMaterial(self):
+        '''
+        Display the currently chosen material
+        '''
+        self._updateButtonColour()
+        self._updateAlpha()
+        self._updateShininess()
+        self._previewMaterial()
+        self._buildTextureComboBox()
+        self._buildRegionComboBox()
+
+    def _updateButtonColour(self):
+        if self._currentMaterial:
+            ambientColour, diffuseColour, emissionColour, specularColour = self._getCurrentMaterialColour()
+        else:
+            print("No Current Material")
+            ambientColour, diffuseColour, emissionColour, specularColour = QtGui.QColor(0,0,0),QtGui.QColor(0,0,0),QtGui.QColor(0,0,0),QtGui.QColor(0,0,0)
+        self._ui.ambientSelectColour_button.setStyleSheet("background-color: {}".format(ambientColour.name()))
+        self._ui.diffuseSelectColour_button.setStyleSheet("background-color: {}".format(diffuseColour.name()))
+        self._ui.emittedSelectColour_button.setStyleSheet("background-color: {}".format(emissionColour.name()))
+        self._ui.specularSelectColour_button.setStyleSheet("background-color: {}".format(specularColour.name()))
+
+    def _getCurrentMaterialColour(self):
+        result, ambientColour = self._currentMaterial.getAttributeReal3(Material.ATTRIBUTE_AMBIENT)
+        result, diffuseColour = self._currentMaterial.getAttributeReal3(Material.ATTRIBUTE_DIFFUSE)
+        result, emissionColour = self._currentMaterial.getAttributeReal3(Material.ATTRIBUTE_EMISSION)
+        result, specularColour = self._currentMaterial.getAttributeReal3(Material.ATTRIBUTE_SPECULAR)
+        intAmbientColour = QtGui.QColor(int(ambientColour[0]*255), int(ambientColour[1]*255), int(ambientColour[2]*255))
+        intDiffuseColour = QtGui.QColor(int(diffuseColour[0]*255), int(diffuseColour[1]*255), int(diffuseColour[2]*255))
+        intEmissionColour = QtGui.QColor(int(emissionColour[0]*255), int(emissionColour[1]*255), int(emissionColour[2]*255))
+        intSpecularColour = QtGui.QColor(int(specularColour[0]*255), int(specularColour[1]*255), int(specularColour[2]*255))
+        return intAmbientColour, intDiffuseColour, intEmissionColour, intSpecularColour
+
+    def _selectColourClicked(self, event):
+        color = QtWidgets.QColorDialog.getColor()
```

Comment:
The `QColorDialog` needs this widget as a parent so that it is owned by something.  The colour should also be initialised with the colour of the sender.  You can get this using the `sender()` method.